libraries/protocol/include/graphene/protocol/custom_authority.hpp

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2019 Contributors.
+ *
+ * The MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+#pragma once
+#include <graphene/protocol/base.hpp>
+#include <graphene/protocol/asset.hpp>
+#include <graphene/protocol/authority.hpp>
+#include <graphene/protocol/restriction.hpp>
+
+namespace graphene { namespace protocol {
+
+   /**
+    * @brief Create a new custom authority
+    * @ingroup operations
+    */
+   struct custom_authority_create_operation : public base_operation {
+      struct fee_parameters_type {
+         uint64_t basic_fee = GRAPHENE_BLOCKCHAIN_PRECISION;
+         uint32_t price_per_byte = GRAPHENE_BLOCKCHAIN_PRECISION / 10;
```

Comment:
The entire `custom_authority_create_operation` used in the `custom_auths` test (line 180) is 54 bytes, and it has an attribute assertion with two sub-restrictions. With these numbers, that's a 4.3 BTS fee. I wouldn't be opposed to making them a bit lower, but I also think this feature deserves a nontrivial fee.

We do use `price_per_kb` in other places, but IMO that gives too little precision. I chose `price_per_byte` to get better precision. We can't really get any higher precision than `price_per_byte`...

`price_per_byte` is 32 bits because I figured that was big enough, and it's intended to be a multiplier. I can make it 64 if that's preferable.

Reply:
Thanks. I think you can leave it as is.